Moonshot/player/player.gd

```diff
@@ -73,15 +70,28 @@ func _physics_process(_delta) -> void:
 	get_node("TurnAxis").rotation = PI + (position + get_node("TurnAxis").position).angle_to_point(get_global_mouse_position())
 
 	if Input.is_action_pressed("shoot") and !cooldown and !safety:
-		var weapon = player_arsenal.get_weapon()
-		var shot = weapon.shoot().instance()
+		var weapon = Utils.player_arsenal.get_weapon()
+		if weapon.ammo == 0:
+			Utils.player_arsenal.set_weapon('laser_blaster')
+			weapon = Utils.player_arsenal.get_weapon()
+		var shots = weapon.shoot()
 
 		cooldown = true
+		
+		for i in range(len(shots)):
+			var shot = shots[i]
+			var shot_rotation_modifier = weapon.shot_rotation_modifiers[i]
+			shot.position = get_node("TurnAxis/CastPoint").get_global_position()
+			shot.rotation = get_node("TurnAxis").rotation 
+			var random_spread
+			if "shot_spread" in weapon:
+				random_spread = rand_range(-weapon.shot_spread, weapon.shot_spread)
+			else:
+				random_spread = 0
+			shot.rotation += ((shot_rotation_modifier + random_spread)*2*PI) / 360 
+			get_parent().add_child(shot)
+		
```

Comment:
I'm not sure if we should be getting weapon  every shot, its an attribute that is not changing very often